docs/proposals/06-immutable-etcd-backups.md

```diff
@@ -0,0 +1,258 @@
+  // Schedule defines a cron schedule (e.g., "0 */6 * * *").
+  Schedule *string `json:"schedule,omitempty"`
+
+  // GarbageCollectionConfig specifies the configuration for snapshot GC.
+  GarbageCollectionConfig *GarbageCollectionConfig `json:"garbageCollectionConfig,omitempty"`
+}
+
+type GarbageCollectionConfig struct {
+  // GarbageCollectionPolicy (e.g., "LimitBased" or "Exponential").
+  GarbageCollectionPolicy *string `json:"garbageCollectionPolicy,omitempty"`
+  // MaxBackupsLimitBasedGC sets the maximum number of full snapshots to keep.
+  MaxBackupsLimitBasedGC *int32 `json:"maxBackupsLimitBasedGC,omitempty"`
+  // DeltaSnapshotRetentionPeriod indicates how long to keep delta snapshots (e.g., "72h").
+  DeltaSnapshotRetentionPeriod *metav1.Duration `json:"deltaSnapshotRetentionPeriod,omitempty"`
+}
+```
+
+**Sample YAML**:
+
+```yaml
+spec:
+  config:
+    schedule: "0 0 * * *"
+    garbageCollectionConfig:
+      garbageCollectionPolicy: "LimitBased"
+      maxBackupsLimitBasedGC: 5
+      deltaSnapshotRetentionPeriod: "72h"
+```
+
+## Compatibility
+
+These changes are compatible with existing etcd clusters and current backup processes.
+
+- **Backward Compatibility:**
+  - Clusters without immutable buckets continue to function without any changes.
+- **Forward Compatibility:**
+  - Clusters can opt in to use immutable backups by configuring the bucket accordingly (as described in [Configuring Immutable Backups](#configuring-immutable-backups)) and setting `etcd.spec.backup.store.immutability.retentionType == "Bucket"`.
+  - The enhanced hibernation logic in the etcd controller is additive, meaning it does not interfere with existing workflows.
+
+### Impact for Operators
+
+In scenarios where you want to exclude certain snapshots from an etcd restore, you previously could simply delete them from object storage. However, when bucket-level immutability is enabled, deleting existing immutable snapshots is no longer possible. To address this need, most cloud providers allow adding **custom annotations or tags** to objects—even immutable ones—so they can be logically excluded without physically removing them.
+
+`etcd-backup-restore` supports ignoring snapshots based on annotations or tags, rather than deleting them. Operators can add the following key-value pair to any snapshot object to exclude it from future restores:
+
+- **Key:** `x-etcd-snapshot-exclude`  
+- **Value:** `true`  
+
+Because these tags or annotations do not modify the underlying snapshot data, they are permissible even for immutable objects. Once these annotations are in place, `etcd-backup-restore` will detect them and skip the tagged snapshots during restoration, thus preventing unwanted snapshots from being used. For more details, see the [Ignoring Snapshots during Restoration](https://github.com/gardener/etcd-backup-restore/blob/master/docs/usage/enabling_immutable_snapshots.md#ignoring-snapshots-during-restoration).
```

Comment:
Please add a `limitation` note for AWS, since we currently don't support this for S3 buckets, atleast at the time of writing this proposal.

Reply:
Added [here](https://github.com/gardener/etcd-druid/pull/884/commits/6e4511a5afcb65ebe8ef391826b7ec1769411853#diff-981f3bcbeed277184597de9af59f3e40fa0c76a43f84c739a0d17fd3948d2f8aR285-R287)

Reply:
Thank you